ETL/Python ETL/MatchHistoryETL.py

```diff
@@ -0,0 +1,210 @@
+                               "AvgH": "MarketAvgHomeTeam",
+                               "AvgD": "MarketAvgDraw",
+                               "AvgA": "MarketAvgAwayTeam",
+                               "B365>2.5": "B365Over2.5Goals",
+                               "B365<2.5": "B365Under2.5Goals",
+                               "Max>2.5": "MarketMaxOver2.5Goals",
+                               "Max<2.5": "MarketMaxUnder2.5Goals",
+                               "Avg>2.5": "MarketAvgOver2.5Goals",
+                               "Avg<2.5": "MarketAvgUnder2.5Goals"},
+                   inplace = True)
+
+
+# Add MatchID column
+
+website_frame.insert(0, "MatchID", constants.CURRENT_SEASON_TAG + "_" + website_frame["HomeTeam"] + "_" + website_frame["AwayTeam"])
+
+# Add season column
+
+website_frame.insert(1, "Season", constants.CURRENT_SEASON_TAG)
+
+
+def calculate_matchweek(cursor):
+    """
+        Finds out last registered matchweek for current season in Database. 
+        Returns last registerered matchweek plus one. If there's any problem 
+        with databse or season has not even started, then returns default matchweek "1"
+    """
+    try:
+        matchweek_query = f"SELECT MAX(\"MatchWeek\") FROM public.match_history WHERE \"Season\" = '{constants.CURRENT_SEASON_TAG}'"
+        cursor.execute(matchweek_query)
+        matchweek_result = cursor.fetchone()
+        if matchweek_result[0] is None:
+            return constants.DEFAULT_MATCHWEEK
+        else:
+            return matchweek_result[0] + 1
+    except psycopg2.Error as e:
+        print ("An error ocurred in database")
+        print (e)
+        return constants.DEFAULT_MATCHWEEK
+
+
+# Stablish a connection to Database data source and fetch last game so we can know current matchweek
+
+try:
+    connection = psycopg2.connect(
+        host = constants.DB_SERVER,
+        port = constants.DB_PORT,
+        user = constants.DB_USER,
+        password = constants.DB_PASSWORD,
+        database = constants.DB_NAME
+    )
+except psycopg2.Error as e:
+    print (f'Can not connect to the postgress database "{constants.DB_NAME}". Make sure database server is running')
+    print (e)
+else:
+    print (f'Connection to database "{constants.DB_NAME}" stablished. Listening at port {constants.DB_PORT}')
+
+
+# Find out current season matchweek
+cursor = connection.cursor()
+next_matchweek = calculate_matchweek(cursor)
+
+
+# Add MatchWeek column
+
+website_frame.insert(2, "MatchWeek", next_matchweek)
+
+# Add Points columns
+
+conditions = [
+     website_frame["FullTimeResult"] == 'H',
+     website_frame["FullTimeResult"] == 'D',
+     website_frame["FullTimeResult"] == 'A'
+]
+
+home_points = [ 3, 1, 0]
+away_points = [ 0, 1, 3]
+
+website_frame["HomeTeamPoints"] = np.select(conditions, home_points)
+website_frame["AwayTeamPoints"] = np.select(conditions, away_points)
+
+
+def generate_upsert_stmn(table_name, conflict_column, dataframe):
```

Comment:
I designed my own function to generate a custom INSERT ON CONFLICT statement , not sure if this is the best way of doing it, as some comments on internet say writting your own SQL Statements is not secure, your code is exposed to SQL injection

Reply:
I decided to remove this function, since there's better options in other libraries like psycopg2 that are efficient and safe to use.